fix: Wallet solution editorial changes (#387)
italia/eidas-it-wallet-docs: docs/en/wallet-attestation.rst

Requirements
------------

> The following requirements for the Wallet Attestation must be met:

Reviewer: since we use here must and then within each requirement other language like should or may, I suggest to reformulate the sentence

```suggestion
The requirements for the Wallet Attestation are defined below:
```

> The following requirements for the Wallet Attestation must be met:

Reviewer 2: ```suggestion
The following requirements apply to the Wallet Attestation:
```

- The Wallet Attestation MAY be used multiple times during its validity period, allowing for repeated authentication and authorization without the need to request new attestations with each interaction.
- The Wallet Attestation MUST be short-lived and MUST have an expiration date/time, after which it SHOULD no longer be considered valid.
- The Wallet Attestation MUST NOT be issued by the Wallet Provider if the authenticity, integrity, and genuineness are not guaranteed. In this case, the Wallet Instance MUST be revoked.
- Each Wallet Instance SHOULD be able to request multiple attestations with different ephemeral public keys associated with them. This requirement provides a privacy-preserving measure, as the public key MAY be used as a tracking tool during the presentation phase (see also the point listed below).
- The Wallet Attestation MUST NOT contain any information that can be used to directly identify the User.
- The Wallet Instance MUST secure a Wallet Attestation as a prerequisite for transitioning to the Operational state, as defined by `ARF`_.
- Private keys MUST be generated and stored in the WSCD using at least one of the approaches listed below:

>   - **Local Internal WSCD**: In this approach, the WSCD relies entirely on the device's native cryptographic hardware, such as the Secure Enclave on iOS devices or the Hardware-Backed Keystore or Strongbox on Android devices.

Reviewer 2: ```suggestion
  - **Local Internal WSCD**: The WSCD relies entirely on the device's native cryptographic hardware, such as the Secure Enclave on iOS devices or the Hardware-Backed Keystore or Strongbox on Android devices.
```

  - **Local External WSCD**: The WSCD is hardware external to the User's device, such as a smart card compliant with *GlobalPlatform* and supporting *JavaCard*.

>   - **Remote WSCD**: Here, the WSCD utilizes a remote Hardware Security Module (HSM).

Reviewer 2: ```suggestion
  - **Remote WSCD**: The WSCD utilizes a remote Hardware Security Module (HSM).
```

**Step 6**: The Wallet Instance, through the operating system, creates a pair of Cryptographic Hardware Keys and stores the corresponding Cryptographic Hardware Key Tag in local storage once the following requirements are met:

   1. It MUST ensure that Cryptographic Hardware Keys do not already exist. If they do exist and the Wallet is in the initialization phase, they MUST be deleted.
   2. It MUST generate a pair of asymmetric Elliptic Curve keys (Cryptographic Hardware Keys) via a local WSCD.
   3. It SHOULD obtain a unique identifier (Cryptographic Hardware Key Tag) for the generated Cryptographic Hardware Keys from the operating system. If the operating system permits specifying a tag during the creation of keys, then a random string for the Cryptographic Hardware Key Tag MUST be selected. This random value MUST be collision-resistant and unpredictable to ensure security. To achieve this, consider using a cryptographic hash function or a secure random number generator provided by the operating system or a reputable cryptographic library.
   4. If the previous points are satisfied, it MUST store the Cryptographic Hardware Key Tag in local storage.

.. note::

>     **WSCD**: The Wallet Instance MAY use a local WSCD for key generation on devices that support this feature. On Android devices, Strongbox is RECOMMENDED; Trusted Execution Environment (TEE) SHOULD be used only when Strongbox is unavailable. For iOS devices, Secure Elements (SE) SHOULD be used. Given that each OEM offers a distinct SDK for accessing the local WSCD, the discussion hereafter will address this topic in a general context.

Reviewer 2: from the current text, it seems that if the strongbox is available, the wallet instance may use TEE. Is this what we want?

Author: `Strongbox is RECOMMENDED; Trusted Execution Environment (TEE) SHOULD be used only when Strongbox is unavailable`

Author: What doesn't make sense to you about this sentence?

Reviewer: @grausof use MAY instead of SHOULD, giving for now all the responsability to the implementers ...

nobody should use that unless there are strong motivation to do that

Reviewer 2: this is clear. my point is: if both strongbox and tee are available the latter may be used (as we say RECOMMENDED). On the basis of what is tee chosen rather than strongbox?
